OrbitGl/CallStackDataView.cpp

```diff
@@ -105,27 +96,42 @@ void CallStackDataView::OnFilter(const std::string& a_Filter) {
 }
 
 //-----------------------------------------------------------------------------
-Function& CallStackDataView::GetFunction(unsigned int a_Row) {
-  static Function dummy;
+void CallStackDataView::OnDataChanged() {
+  size_t numFunctions = m_CallStack ? m_CallStack->m_Depth : 0;
+  m_Indices.resize(numFunctions);
+  for (size_t i = 0; i < numFunctions; ++i) {
+    m_Indices[i] = i;
+  }
+}
 
-  if (m_CallStack) {
-    if (a_Row < m_CallStack->m_Depth) {
-      ScopeLock lock(Capture::GTargetProcess->GetDataMutex());
-
-      DWORD64 addr = m_CallStack->m_Data[a_Row];
-      Function* func =
-          Capture::GTargetProcess->GetFunctionFromAddress(addr, false);
-
-      if (func) {
-        return *func;
-      } else if (Capture::GSamplingProfiler) {
-        dummy.SetPrettyName(
-            ws2s(Capture::GSamplingProfiler->GetSymbolFromAddress(addr)));
-        dummy.SetAddress(addr);
-        return dummy;
-      }
-    }
+//-----------------------------------------------------------------------------
+Function* CallStackDataView::GetFunction(int a_Row) {
+  if (m_CallStack != nullptr && a_Row < m_CallStack->m_Depth &&
+      Capture::GTargetProcess != nullptr) {
+    ScopeLock lock(Capture::GTargetProcess->GetDataMutex());
+
+    uint64_t address = m_CallStack->m_Data[a_Row];
+    Function* function =
+        Capture::GTargetProcess->GetFunctionFromAddress(address, false);
+    return function;
+  }
+  return nullptr;
+}
+
+//-----------------------------------------------------------------------------
+Function& CallStackDataView::GetFunctionOrDummy(int a_Row) {
+  Function* function = GetFunction(a_Row);
+  if (function != nullptr) {
+    return *function;
   }
 
+  static Function dummy;
```

Comment:
This is not really a dummy - you are populating it with correct name and address. Seems like this might need some lightweight function representation which could be populated from Function or filled up on the fly, and it can be temporary - which will help with static here. WDYT?

Reply:
I'd be in favor of removing the static dummy here.  Maybe Process could create functions in void AddSymbol(uint64_t a_Address, std::shared_ptr<LinuxSymbol> a_Symbol).  I wouldn't treat them as regular functions however as their address is not the start of a function.  But we could then ask the process for these "dummy" functions in the case where we didn't find a fully fledged function.

Reply:
That being said, I think we should get this PR in asap to fix the crash and make non-trivial improvements in other PRs.

Reply:
In principle I agree, but I wouldn't spend time rethinking this now. For example, as I wrote in the commit message, we might want a CallStackDataView that doesn't inherit from FunctionsDataView if we want the two tables to be different, but this would need more thought.

Reply:
That's fair.